src/commands/email/send.js

```diff
@@ -8,27 +8,86 @@ const path = require('path');
-      this.logger.error('Make sure you have an environment variable called SENDGRID_API_KEY set up with your SendGrid API key. Visit https://app.sendgrid.com/settings/api_keys to get an API key.');
+      this.logger.error(
+        'Make sure you have an environment variable called SENDGRID_API_KEY set up with your SendGrid API key. Visit https://app.sendgrid.com/settings/api_keys to get an API key.'
+      );
       return this.exit(1);
     }
+
+    const pipedInput = await this.readStream();
+    if (pipedInput) {
+      this.processData(pipedInput);
+    }
+
+    this.isTTY = process.stdin.isTTY || this.flags['force-tty'];
+
     await this.promptForFromEmail();
-    const validFromEmail = this.validateEmail(this.fromEmail);
     await this.promptForToEmail();
-    const validToEmail = this.validateEmail(this.toEmail);
     await this.promptForSubject();
     await this.promptForText();
-    const sendInformation = { to: validToEmail, from: validFromEmail[0], subject: this.subjectLine, text: this.emailText, html: '<p>' + this.emailText + '</p>' };
-    const attachmentVerdict = await this.askAttachment();
-    await this.promptAttachment(attachmentVerdict);
-    if (this.attachment) {
-      const fileContent = this.readFile(this.attachment);
-      const attachment = this.createAttachmentArray(fileContent);
+
+    if (!this.isTTY && (!this.flags.to || !this.flags.text || !this.subjectLine || !this.fromEmail)) {
+      this.logger.error('All flags must be provided to send email.');
+      return this.exit(1);
+    }
+
+    const validFromEmail = this.validateEmail(this.fromEmail);
+    const validToEmail = this.validateEmail(this.toEmail);
+
+    const sendInformation = {
+      to: validToEmail,
+      from: validFromEmail[0],
+      subject: this.subjectLine,
+      text: this.emailText,
+      html: '<p>' + this.emailText + '</p>'
+    };
+
+    if (this.pipedInfo) {
+      const attachment = this.createAttachmentArray(this.pipedInfo);
       sendInformation.attachments = attachment;
+    } else {
+      const attachmentVerdict = await this.askAttachment();
+      await this.promptAttachment(attachmentVerdict);
+      if (this.attachment) {
+        const fileContent = this.readFile(this.attachment);
+        const attachment = this.createAttachmentArray(fileContent);
+        sendInformation.attachments = attachment;
+      }
     }
     await this.sendEmail(sendInformation);
   }
 
+  myGetStdin() {
+    const { stdin } = process;
+    let result = '';
+
+    return new Promise(resolve => {
+      if (stdin.isTTY) {
+        resolve(result);
+        return;
+      }
+
+      stdin.setEncoding('utf8');
+
+      stdin.once('data', data => {
+        resolve(data);
+      });
+    });
+  }
+
+  async readStream() {
+    const input = await this.myGetStdin();
+    const base64data = Buffer.from(input).toString('base64');
+    return base64data;
+  }
+
+  processData(input) {
+    this.fileName = 'piped.txt';
```

Comment:
It's unclear why this is being hard coded. This is _just_ the name that the file will have when it's sent? Is this worth a comment?